frontend/packages/operator-lifecycle-manager/src/components/operand.tsx

```diff
@@ -167,8 +167,57 @@ export const OperandTableHeader = () => {
   ];
 };
 
+export enum OperatorStatusType {
+  phase = 'phase',
+  status = 'status',
+  state = 'state',
+  conditions = 'conditions',
+}
+
+export const OperatorStatusTypeText = {
+  [OperatorStatusType.phase]: 'Phase:',
+  [OperatorStatusType.status]: 'Status:',
+  [OperatorStatusType.state]: 'State:',
+  [OperatorStatusType.conditions]: 'Condition:',
+};
+
+export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
+  statusObject,
+}) => {
+  let iconAndText = <div className="text-muted">Unknown</div>;
+  if (_.isEmpty(statusObject)) {
+    return iconAndText;
+  }
+  _.forEach(OperatorStatusType, (key: string) => {
+    if (_.has(statusObject, key)) {
+      const status = statusObject[key];
+      if (!_.isEmpty(status)) {
+        let statusIcon;
+        if (key !== OperatorStatusType.conditions) {
+          if (status === 'Running') {
+            statusIcon = <SuccessStatus title={status} />;
+          } else {
+            statusIcon = <Status status={status} />;
+          }
+        } else if (_.has(status[0], 'type')) {
```

Comment:
Don't you need to check the true/false value of the condition?

Reply:
@spadgett , _.has will return true if the path exists and false otherwise. Is that what you . mean? Also, status[0] is available since _.isEmpty was false.